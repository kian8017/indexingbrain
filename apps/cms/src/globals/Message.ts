import { GlobalConfig } from "payload/types";

const Message: GlobalConfig = {
  slug: "message",
  fields: [
    {
      name: "message",
      type: "text",
      required: true,
    },
  ],
  access: {
    read: () => true,
  },
  admin: {
    description:
      "The pop-up message shown to people when visiting the search page.",
  },
};

export default Message;
