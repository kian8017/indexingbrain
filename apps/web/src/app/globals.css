@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  h1 {
    @apply text-3xl font-bold;
  }
  h2 {
    @apply text-2xl font-semibold;
  }
  h3 {
    @apply text-xl;
  }
  h4 {
    @apply text-lg;
  }
  h5 {
    @apply text-base;
  }
  h6 {
    @apply text-sm;
  }

  hr {
    @apply mx-16 my-4 border-2 border-secondary/50 rounded;
  }
  p {
    @apply text-xl;
  }
  main ul {
    @apply ml-4 list-inside list-disc text-xl;
  }
  main ol {
    @apply ml-4 list-inside list-decimal text-xl;
  }
  main a {
    @apply text-xl underline link link-secondary link-hover visited:link-accent;
  }
}

:root,
html,
body {
  top: 0;
  bottom: 0;
  min-height: 100vh;
  width: 100%;
}
