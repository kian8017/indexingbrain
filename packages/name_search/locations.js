import { readdir } from "fs/promises";

async function parseLocations(rootFolder) {
  const fsEntries = await readdir(rootFolder, { withFileTypes: true });
  return fsEntries
    .filter((entry) => entry.isDirectory())
    .map((entry) => {
      const pieces = entry.name.split(" ");
      const abbr = pieces[0];
      const name = pieces.slice(1).join(" ");
      return {
        name,
        abbr,
        folder: entry.name,
      };
    });
}

export { parseLocations };
