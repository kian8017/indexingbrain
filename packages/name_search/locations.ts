import { readdir } from "fs/promises";
import { join } from "path";

type Location = {
  name: string;
  abbr: string;
  path: string;
};

async function parseLocations(rootFolder: string): Promise<Location[]> {
  const fsEntries = await readdir(rootFolder, { withFileTypes: true });
  return fsEntries
    .filter((entry) => entry.isDirectory())
    .map((entry) => {
      const pieces = entry.name.split(" ");
      const abbr = pieces[0];
      const name = pieces.slice(1).join(" ");
      return {
        name,
        abbr,
        path: join(rootFolder, entry.name),
      };
    });
}

export { parseLocations };
