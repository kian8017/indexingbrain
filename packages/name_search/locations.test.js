import { describe, it, expect } from "@jest/globals";
import { parseLocations } from "./locations";
import { join } from "path";


describe("locations", () => {
  it("should correctly parse the test folder", async () => {
    const testFolder = new URL("test", import.meta.url);
    const locations = await parseLocations(testFolder);
    expect(locations).toMatchSnapshot();
  });
});
