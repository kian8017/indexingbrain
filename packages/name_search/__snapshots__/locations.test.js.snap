// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`locations should correctly parse the test folder 1`] = `
[
  {
    "abbr": "AFG",
    "folder": "AFG Afghanistan",
    "name": "Afghanistan",
  },
  {
    "abbr": "BZ",
    "folder": "BZ Brazil",
    "name": "Brazil",
  },
  {
    "abbr": "SPSH",
    "folder": "SPSH Spanish",
    "name": "Spanish",
  },
]
`;
